### PR TITLE
Support Windows ARM

### PR DESCRIPTION
**Pull request by ivan-diachenko-td**
In this changeset, we add a dependency to Windows ARM binaries since we recently [introduced](https://github.com/TeamDev-IP/JxBrowser/issues/2990) the support of this architecture. As for now, this will only be supported in JxBrowser 8.

Also, I added a check for some artifacts supported only in JxBrowser 8. From now on, if the user decides to use the `jxbrowser-kotlin` artifact with JxBrowser 7, for example, the build will fail.

### CODE REVIEW

File: src/main/kotlin/com/teamdev/jxbrowser/gradle/JxBrowserExtension.kt [hunk 1]
```diff
@@ -112,6 +112,14 @@ public open class JxBrowserExtension(private val project: Project) {
      */
     public val win32: Provider<String> = artifact("win32")
 
+    /**
+     * Returns a dependency notation for the `jxbrowser-win64-arm`,
+     * an artifact with Chromium Windows ARM 64-bit binaries.
+     *
+     * As for now, Windows ARM is only supported in JxBrowser 8.x.x.
+     */
+    public val win64Arm: Provider<String> = artifact("win64-arm")
```
Comment (Danil-Didkovskiy) [inline src/main/kotlin/com/teamdev/jxbrowser/gradle/JxBrowserExtension.kt:121]: How about just `winArm`?

File: src/main/kotlin/com/teamdev/jxbrowser/gradle/JxBrowserExtension.kt [hunk 2]
```diff
@@ -169,6 +181,9 @@ public open class JxBrowserExtension(private val project: Project) {
     private fun artifact(shortName: String) =
         project.providers.provider {
             check(version.isNotBlank()) { "JxBrowser version is not specified." }
+            check(!(!version.startsWith("8") && shortName.equals("win64-arm"))) {
```
Comment (Danil-Didkovskiy) [inline src/main/kotlin/com/teamdev/jxbrowser/gradle/JxBrowserExtension.kt:184]: Let's simplify it to make it easier to read:

```kotlin
if (shortName == "win64-arm" && !version.startsWith("8")) {
    error("Windows ARM is not supported in JxBrowser $version. Use 8.x.x.")
}
```

File: src/main/kotlin/com/teamdev/jxbrowser/gradle/JxBrowserExtension.kt [hunk 3]
```diff
@@ -150,14 +158,18 @@ public open class JxBrowserExtension(private val project: Project) {
...
+                { isX64Bit() && isWindows() } to win64,
+                { isX64Bit() && isLinux() } to linux64,
+                { isX64Bit() && isMac() } to mac,
+                { is32Bit() && isWindows() } to win32,
+                { isArm() && isLinux() } to linuxArm,
+                { isArm() && isMac() } to macArm,
             )
+        if (version.startsWith("8")) {
+            // As for now, Windows ARM is only supported in JxBrowser 8.
+            platformMap.put({ isArm() && isWindows() }, win64Arm)
```
Comment (Danil-Didkovskiy) [inline src/main/kotlin/com/teamdev/jxbrowser/gradle/JxBrowserExtension.kt:171]: I think we should always add this entry to `platformMap`. If the current version is not 8, then we will throw an exception later inside the `artifact()` method with a clear message to the user.

File: src/test/kotlin/com/teamdev/jxbrowser/gradle/JxBrowserPluginFunctionalTest.kt [hunk 4]
```diff
@@ -94,7 +94,10 @@ internal class JxBrowserPluginFunctionalTest {
                 "jxbrowser-mac-$jxBrowserVersion.jar",
                 "jxbrowser-mac-arm-$jxBrowserVersion.jar",
             )
-
+        if (jxBrowserVersion.startsWith("8")) {
```
Comment (Danil-Didkovskiy) [inline src/test/kotlin/com/teamdev/jxbrowser/gradle/JxBrowserPluginFunctionalTest.kt:97]: I don't think it's necessary to test that here. We can cover it in the test below, which specifically tests the JxBrowser 8 jars.

**Review by Danil-Didkovskiy — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/teamdev/jxbrowser/gradle/JxBrowserExtension.kt (hunk 1 above)
Comment (ivan-diachenko-td) [reply src/main/kotlin/com/teamdev/jxbrowser/gradle/JxBrowserExtension.kt:121]: For the sake of saving consistency with `win64`/`win32` let's use this naming.

**Comment by ivan-diachenko-td** [comment]
@Danil-Didkovskiy PTAL

**Review by Danil-Didkovskiy — APPROVED** [review]

**Pull request merged**